lib/exponential-backoff/exponential-backoff.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2024 Attila Szakacs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#ifndef EXPONENTIAL_BACKOFF_H
+#define EXPONENTIAL_BACKOFF_H
+
+#include "syslog-ng.h"
+
+#define EXPONENTIAL_BACKOFF_OPTIONS_INITIAL_SECONDS_DEFAULT (0.1)
+#define EXPONENTIAL_BACKOFF_OPTIONS_MAXIMUM_SECONDS_DEFAULT (1)
```

Comment:
This seems very low in my opinion compared to the old default. I think we shouldn't go lower than 10 seconds by default. If we are planning to introduce this for traditional syslog/network drivers too, I would also consider going with different defaults there (or just keep the old 60 seconds as maximum?).

Note: connection attempts are logged internally on a high log level. For example, using the HTTP produces 2 error messages with each reopen attempt:

```
curl: error sending HTTP request; url='...', error='Couldn\'t connect to server', worker_index='0', driver='#anon-destination0#1', location='...'
Server disconnected while preparing messages for sending, trying again; driver='#anon-destination0#1', location='...', worker_index='0', reopen='0.00', batch_size='1'
```

TLDR: Do we really want `1` as maximum? If so, then we should do something about internal logging connection attempts as we are currently spamming 2-3 error messages every second per worker.
In my opinion, we should rather choose a higher default (60 or 10 would be my suggestion) and keep the logging as-is. The exponential retry mechanism will make sure that we reconnect instantly in case of momentary errors, and we even have 5 more attempts before reaching `1` second of wait time and 10 attempts before reaching `60`.

Reply:
I assumed this was the "default" for max seconds, unless specified otherwise.